app/Http/Controllers/Api/ChannelController.php

```diff
@@ -0,0 +1,359 @@
+<?php
+
+namespace Suyabay\Http\Controllers\Api;
+
+use DB;
+use Validator;
+use Suyabay\Channel;
+use Suyabay\Episode;
+use Suyabay\Http\Requests;
+use League\Fractal\Manager;
+use Illuminate\Http\Request;
+use League\Fractal\Resource\Item;
+use Illuminate\Mail\Mailer as Mail;
+use League\Fractal\Resource\Collection;
+use Illuminate\Support\Facades\Response;
+use Suyabay\Http\Controllers\Controller;
+use Suyabay\Http\Transformers\ChannelTransformer;
+
+class ChannelController extends Controller
+{
+    protected $mail;
+    protected $response;
+    protected $fractal;
+
+    public function __construct(Manager $fractal, Mail $mail)
+    {
+        $this->fractal = $fractal;
+
+        parent::__construct($mail);
+    }
+
+    /**
+     * This method lists all channels
+     * @param $page
+     * @param $request
+     * @param $response
+     *
+     * @return json $response
+     */
+    public function getAllChannels(Request $request, ChannelTransformer $channelTransformer)
+    {
+        $perPage = $request->query('results') ? : 10;
+
+        $channels = Channel::orderBy('channels.id', 'asc')
```

Comment:
**Code Quality- Nitpick**
@andela-tolotin Looking at what you want to achieve here, this query is looking to complex. Lets discuss about this.